core/config/config_base.py

```
@@ -0,0 +1,497 @@
+            Configuration instance
+
+        Raises:
+            ConfigValidationError: If validation fails
+        """
+        try:
+            instance = cls(**data)
+            instance.validate_full()
+            return instance
+        except ValidationError as e:
+            raise ConfigValidationError("Failed to create from dict",
+                                      [f"{err['loc']}: {err['msg']}" for err in e.errors])
```

Ruff (0.14.13)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)